packages/pci-sdk-web/src/iframe.service.ts

```diff
@@ -0,0 +1,176 @@
+import { IAuthOptions, INetworkLogo, Size, Values } from './index';
+import IThemeName from './types/IThemeName';
+
+interface IInitIframeArgs {
+	allowedCorsDomain: string;
+	auth: IAuthOptions;
+	debug?: boolean;
+	networkLogo?: INetworkLogo;
+	element?: HTMLElement;
+	size?: Size;
+	theme?: IThemeName;
+	url: string;
+	values?: Values;
+}
+
+export async function initIframe(
+	args: IInitIframeArgs
+): Promise<HTMLIFrameElement> {
+	const pciElement = args.element || document.getElementById('apto-pci-sdk');
+
+	if (!pciElement) {
+		throw new Error('You need to provide an HTML element to init the PCI SDK');
+	}
+
+	const iframeLoadedPromise = _waitForEvent({
+		cors: args.allowedCorsDomain,
+		eventType: 'apto-iframe-ready',
+	});
+
+	const iframeStartedPromise = _waitForEvent({
+		cors: args.allowedCorsDomain,
+		eventType: 'apto-iframe-auth-ready',
+	});
+
+	// Reset content
+	pciElement.innerHTML = '';
+	// Create the iframe
+	const $aptoIframe = _createIframe(args);
+	// Append child to the DOM
+	pciElement.appendChild($aptoIframe);
+	// Wait for the iframe to be loaded
+	await iframeLoadedPromise;
+	// Send auth data via postMessage
+	const messageEvent = JSON.stringify({
+		type: 'apto-iframe-auth',
+		userToken: args.auth.userToken,
+		apiKey: args.auth.apiKey,
+	});
+	$aptoIframe.contentWindow?.postMessage(messageEvent, args.allowedCorsDomain);
+	// Wait for the iframe auth to be started
+	await iframeStartedPromise;
+	// If we are here, the iframe is loaded and was started.
+	return $aptoIframe;
+}
+
+interface IWaitForEventArgs {
+	cors: string;
+	eventType: string;
+}
+
+/**
+ * Promise that is resolved when the right event type comes from the window with the allowed CORS domain.
+ */
+function _waitForEvent(args: IWaitForEventArgs): Promise<void> {
+	return new Promise<void>((resolve) => {
+		window.addEventListener('message', _onMessage, false);
+
+		function _onMessage(event: MessageEvent) {
+			if (!_isMessageAllowed({ event, cors: args.cors })) {
+				return;
+			}
+
+			const message = JSON.parse(event.data);
+
+			if (message.type === args.eventType) {
+				resolve();
+			}
+		}
+	});
+}
+
+interface IIsMessageAllowedArgs {
+	event: MessageEvent;
+	cors: string;
+}
+
+/**
+ * Returns true if the message is allowed to be received by the window.
+ */
+function _isMessageAllowed(args: IIsMessageAllowedArgs): boolean {
+	if (args.cors === '*') {
+		return true;
+	}
+	return args.event.origin === args.cors;
+}
```

Comment:
Let's pair review thins one 🤔 , I believe if `cors` is enabled, we're losing `window.postMessage` security here?